app/aws/s3.py

```diff
@@ -132,33 +133,74 @@ def cleanup_old_s3_objects():
         )
 
 
-def get_s3_files():
+def get_job_id_from_s3_object_key(key):
+    object_arr = key.split("/")
+    job_id = object_arr[1]  # get the job_id
+    job_id = job_id.replace(".csv", "")  # we just want the job_id
+    return job_id
+
+
+def read_s3_file(bucket_name, object_key, s3res):
+    """
+    This method runs during the 'regenerate job cache' task.
+    Note that in addition to retrieving the jobs and putting them
+    into the cache, this method also does some pre-processing by
+    putting a list of all phone numbers into the cache as well.
+
+    This means that when the report needs to be regenerated, it
+    can easily find the phone numbers in the cache through job_cache[<job_id>_phones]
+    and the personalization through job_cache[<job_id>_personalisation], which
+    in theory should make report generation a lot faster.
+
+    We are moving processing from the front end where the user can see it
+    in wait time, to this back end process.
+    """
+    try:
+        job_id = get_job_id_from_s3_object_key(object_key)
+        if job_cache.get(job_id) is None:
+            object = (
+                s3res.Object(bucket_name, object_key)
+                .get()["Body"]
+                .read()
+                .decode("utf-8")
+            )
+            # TODO we probably don't need this phone number check anymore.
+            # Originally, there were csv uploads that were not valid
+            # for messaging sending.  They may have been experimental, or
+            # they may have not been caught by validation.  When we're sure
+            # all csv files have a phone number column, we can remove the if.
+            # We are essentially just making sure this object looks like a job.
```

Comment:
Thank you for calling this out!  Do you think there's a way we could confidently verify when we reach such a state that we won't need this check anymore?

Reply:
We've had a ticket in 'new' forever ... I just created admin #1974 and assigned to myself.  All we need to do is check that validation rejects uploaded csvs if they don't have a phone number column (or fix validation if they go through).

Reply:
I just tested and got the following red banner error in the UI:

Your file needs a column called ‘phone number’.

So I can remove this comment